### PR TITLE
Blender quaternions

### PR DESCRIPTION
**Pull request by jackcaron**
This PR makes sure that quaternions are used if the rotation_mode == 'QUATERNION' in the exporter.

The only thing not working properly is the cameras. They're not loaded properly with rotationQuaternion and they're always facing a default direction. And with all the different types of cameras, that's something I don't want to risk breaking.

### CODE REVIEW

**Comment by Palmer-JC** [comment]
Can you provide a .blend that exercises these changes?  Anywhere will be fine.  It will eventually end up in the tower of babel directory of the extensions directory.

**Comment by jackcaron** [comment]
https://www.dropbox.com/s/x6k9hbnlne260ct/quatRotation.blend?dl=0

uses quaternions almost everywhere (for things that matters). There's a rotating cube (which work perfectly). The camera is animated, but nothing works with it, but with my modifications, the animation exports properly.

File: src/Loading/Plugins/babylon.babylonFileLoader.ts [hunk 1]
```diff
@@ -478,7 +478,12 @@
                 camera.target = BABYLON.Vector3.FromArray(parsedCamera.target);
             }
         } else {
-            camera.rotation = BABYLON.Vector3.FromArray(parsedCamera.rotation);
+            if (parsedCamera.rotationQuaternion) {
```
Comment (deltakosh) [inline src/Loading/Plugins/babylon.babylonFileLoader.ts]: Warning here: only the followCamera uses the rotationQuaternion. All others cameras only use euler rotation

**Comment by deltakosh** [comment]
I would rather suggest to remove quaternion export for light and camera

**Comment by jackcaron** [comment]
If you have animated rotations, then converting to euler will totally break it. In the Blender exporter, lights don't export animations.

**Comment by deltakosh** [comment]
I completely understand but in this case exporting quaternion animations for cameras won't work as rotationQuaternion is not used

**Comment by bhouston** [comment]
@deltakosh Just a quick note that Quaternion export of cameras is required if we are going to support arbitrary camera paths accurately -- the issue in many tools is that the resulting rotation of an object is composed of a series of rotations that if represented as Euler rotation key will not interpolate correctly between the keys -- we will suffer from gimbal lock style issues, or just plain incorrect rotation interpolation.

**Comment by bhouston** [comment]
I'd suggest that Babylon.js represent rotations intrinsically as Quaternions on Cameras and Lights and offer an Euler like interface to users for easy control.  I've done this a few times on past projects and it works great.

**Comment by deltakosh** [comment]
I totally agree :) This is more a question of doing it

**Comment by deltakosh** [comment]
I'll validate this but I will remove the camera.rotationQuaternion stuff

**Pull request merged**